firebase-firestore/src/androidTest/java/com/google/firebase/firestore/ValidationTest.java

```diff
@@ -461,44 +461,49 @@ public void queriesCannotBeSortedByAnUncommittedServerTimestamp() {
     TaskCompletionSource<Void> offlineCallbackDone = new TaskCompletionSource<>();
     TaskCompletionSource<Void> onlineCallbackDone = new TaskCompletionSource<>();
 
-    collection.addSnapshotListener(
-        (snapshot, error) -> {
-          assertNotNull(snapshot);
-
-          // Skip the initial empty snapshot.
-          if (snapshot.isEmpty()) return;
-
-          assertThat(snapshot.getDocuments()).hasSize(1);
-          DocumentSnapshot docSnap = snapshot.getDocuments().get(0);
-
-          if (snapshot.getMetadata().hasPendingWrites()) {
-            // Offline snapshot. Since the server timestamp is uncommitted, we shouldn't be able to
-            // query by it.
-            assertThrows(
-                IllegalArgumentException.class,
-                () ->
-                    collection
-                        .orderBy("timestamp")
-                        .endAt(docSnap)
-                        .addSnapshotListener((snapshot2, error2) -> {}));
-            offlineCallbackDone.setResult(null);
-          } else {
-            // Online snapshot. Since the server timestamp is committed, we should be able to query
-            // by it.
-            collection
-                .orderBy("timestamp")
-                .endAt(docSnap)
-                .addSnapshotListener((snapshot2, error2) -> {});
-            onlineCallbackDone.setResult(null);
-          }
-        });
+    ListenerRegistration listenerRegistration =
```

Comment:
This is the change from https://github.com/firebase/firebase-android-sdk/pull/1188